porcupine-core/src/Data/Locations/VirtualFile.hs

```diff
@@ -114,13 +114,19 @@ instance HasDefaultMappingRule (VirtualFile a b) where
 
 -- For now, given the requirement of PTask, VirtualFile has to be a Monoid
 -- because a Resource Tree also has to.
+-- TODO: explain the meaning of (<>). From the implementation,
+-- computing the union of serials and ignoring the location seems pretty
+-- arbitrary.
```

Comment:
Yes it is. This Semigroup/Monoid instances are just so that two tasks that manipulate the same VirtualFile can be used in the same pipeline. If that happens, we just merge their serials, and for everything else the first occurence of the VirtualFile (the one that will be therefore the left operand of `(<>)` takes precedence).

But I came to think maybe treating VirtualFiles as Semigroups might not be the right way to do it.